test/connectivity/connectivity.test.jsx

```diff
@@ -0,0 +1,259 @@
+const Connection = require('mongodb-connection-model');
+const CompassConnectPlugin = require('@mongodb-js/compass-connect');
+const CompassConnectComponent = CompassConnectPlugin.default;
+const activateCompassConnect = CompassConnectPlugin.activate;
+
+const deactivateCompassConnect = CompassConnectPlugin.deactivate;
+
+const TEST_TIMEOUT_MS = require('./.mocharc.json').timeout;
+const {
+  dockerComposeDown,
+  dockerComposeUp
+} = require('./docker-instance-manager');
+
+// Hide react warnings.
+const originalWarn = console.warn.bind(console.warn);
+console.warn = (msg) => (
+  !msg.toString().includes('componentWillReceiveProps')
+  && !msg.toString().includes('componentWillUpdate')
+  && originalWarn(msg)
+);
+
+const delay = promisify(setTimeout);
+const ensureConnected = async(timeout, testIsConnected) => {
+  let connected = await testIsConnected();
+  let timespentTesting = 0;
+  while (!connected) {
+    if (timeout > TEST_TIMEOUT_MS || timespentTesting > TEST_TIMEOUT_MS) {
+      throw new Error('Waited for connection, never happened');
+    }
+    debug(`Testing connectivity at timeout=${timeout}, connected=${connected}`);
+    await delay(timeout);
+    timespentTesting += timeout;
+    timeout *= 2; // Try again but wait double.
+    connected = await testIsConnected();
+  }
+  return connected;
+};
+
+const connectionsToTest = [{
+  title: 'local community standalone',
+  dockerComposeFilePath: 'community/docker-compose.yaml',
+  model: {
+    hostname: 'localhost',
+    port: 27020,
+  },
+  expectedInstanceDetails: {
+    _id: 'localhost:27020',
+    hostname: 'localhost',
+    port: 27020,
+    client: {
+      isWritable: true,
+      isMongos: false,
+    },
+    build: {
+      raw: {
+        version: /4.4.[1-9]+$/
+      }
+    },
+    host: {
+      os: 'Ubuntu',
+      os_family: 'linux',
+      kernel_version: '18.04',
+      arch: 'x86_64'
+    },
+    genuineMongoDB: { isGenuine: true, dbType: 'mongodb' },
+    dataLake: { isDataLake: false, version: null },
+  }
+}];
+
+describe('Connectivity', () => {
+  let appRegistry;
+  let compassConnectStore;
+  connectionsToTest.forEach(connection => {
+    before(() => {
+      dockerComposeUp(connection.dockerComposeFilePath);
+
+      appRegistry = new AppRegistry();
+
+      activateCompassConnect(appRegistry);
+
+      global.hadronApp = {
+        appRegistry
+      };
+
+      const ROLE = {
+        name: 'Status',
+        component: () => (<div id="statusPlugin">Status</div>)
+      };
+      global.hadronApp.appRegistry = appRegistry;
+      global.hadronApp.appRegistry.registerRole('Application.Status', ROLE);
+
+      compassConnectStore = appRegistry.getStore('Connect.Store');
+
+      // Remove all logic around saving and loading stored connections.
+      // NOTE: This is tightly coupled with the store in compass-connect.
+      // https://github.com/mongodb-js/compass-connect/blob/master/src/stores/index.js
+      compassConnectStore._saveRecent = () => {};
```

Comment:
what happens if we just let this happen? wouldn't this just work?

Reply:
I mean, avoiding mocking and stubbing at this level would make those tests less of the "white-box" type, which i feel would make it more stable.

The main concern I have is that those mocks are relying on internal details of things that are defined quite far from here. Would not expect to break a test about 'compass can connect with tls' if I refactor the way recent connections are stored.

Reply:
Without stubbing this `saveRecent_` function we save the resulting connection into a folder called `Connections` at the root of Compass. I think it'll take a bit more work to either clean up this folder post run, or even better I think we can use this saved connection as a way to improve our test (by ensuring it saves the connection in the correct format, and maybe even loads it in afterwords). If it sounds good to you I'd like to break this into another ticket so we can merge what we have here and keep working on some other connections, since I think that should be more of an isolated change.
Here's a ticket for it I added to this project:  COMPASS-4630